### ISSUE TITLE
 JsonConvert.SerializeObject returns {} on serializing JSONPatchDocument starting with 11.0.1

### ISSUE THREAD

**Issue by bruceburge** [JamesNK/Newtonsoft.Json#1641]
### Expected behavior

using this as a base example 
https://www.visualstudio.com/en-us/docs/integrate/api/wit/samples#create-bug
(.NET Client Library example) 
with 10.0.2 and 10.0.3 the example object is serialized correctly, and passed along, to vsts.

Starting with 11.0.1 and attempt to serialize JSONPatchDocument  results in {}

Repo steps can be done with a simplified code sample based on  visual studios site, as provided below.
```C#
	JsonPatchDocument patchDocument = new JsonPatchDocument();

	//add fields to your patch document
	patchDocument.Add(
		new JsonPatchOperation() {
		Operation = Operation.Add,
		Path = "/fields/System.Title",
		Value = "Authorization Errors"
		}
	);

	patchDocument.Add(
		new JsonPatchOperation() {
		Operation = Operation.Add,
		Path = "/fields/Microsoft.VSTS.TCM.ReproSteps",
		Value = "Our authorization logic needs to allow for users with Microsoft accounts (formerly Live Ids) - http://msdn.microsoft.com/en-us/library/live/hh826547.aspx"
		}
	);

		var foo = JsonConvert.SerializeObject(patchDocument);
		Console.WriteLine(" Dump	: {0}", foo);
		//use rest of sample for more detailed skimming.
```

**Comment by thevintik** [comment]
I confirm this Issue. Minimal, Complete, and Verifiable Example:
```
[DataContract]
public class SingleItem
{
}

[DataContract]
public class MultipleItems : List<SingleItem>
{
}

var s = JsonConvert.SerializeObject(new MultipleItems { new SingleItem() });
```

Using version 10.0.3 result in **[{}]**, while 11.0.1 produces **{}**

**Comment by bruceburge** [comment]
it should be noted that in the example above, using [DataContract] seems to be the deciding factor on the result.
`
public class SingleItem
{
}


public class MultipleItems : List<SingleItem>
{
}

var s = JsonConvert.SerializeObject(new MultipleItems { new SingleItem() });`

with 11.0.1 produces [{}]

**Comment by thevintik** [comment]
This one duplicates #1627

**Comment by bruceburge** [comment]
@thevintik I think it would be closer to #1598 as JsonPatchDocument implements
[System.Runtime.Serialization.DataContractAttribute]

but yes a duplication that looks like it is scheduled to be fixed in 11.0.2 https://github.com/JamesNK/Newtonsoft.Json/commit/c72f785b9c20328af42aa58ac1e99f32c0c0de52
